### ISSUE TITLE
Insufficient error debug output

### ISSUE THREAD

**Issue by speller** [hashicorp/terraform#29479]
I have a provisioner that tries to setup the current directory:

```
resource "null_resource" "admin-provision" {
  provisioner "local-exec" {
    working_dir = "${path.module}/../scripts/"
    command = "command"
  }
```

But it fails with the following error:

```
local-exec provisioner error

  with module.backend.null_resource.admin-provision,
  on ../../main.tf line 180, in resource "null_resource" "admin-provision":
 180:   provisioner "local-exec" 

Error running command 'command': chdir
../../../scripts/: no such file or directory. Output: 

```

It prints out the relative path but what's the context? It's different from the path I set up in the source code. I'm pretty sure that the directory exists on one level up. But I can't debug this because the relative directory is useless here because I don't know to which directory it relates? 

`TF_LOG=trace` says nothing additional about the error.

It would be nice to print additional data in the error message or in the debug log about the absolute directory tried to access or about the current directory from which the relative path was calculated.

**Comment by apparentlymart** [comment]
Hi @speller,

My understanding of this error message is that `${path.module}` expanded to `../../`, which would suggest that this resource is in a module you called using `source = "../../"` or equivalent. Is that incorrect?

Terraform doesn't typically do anything odd with paths here... the current working directory is the directory you ran `terraform` from. The only exception would be if you used `terraform -chdir=...` to instruct Terraform to change directory prior to running its subcommand, in which case the base directory would be the one you explicitly set.

**Comment by speller** [comment]
Yes, the module is from a directory on a higher level. And yes, I'm using `chdir`. In this case, building an absolute directory path in mind is not efficient. An absolute path printed in the error log (even if hidden under TF_LOG) would help a lot.
